### PR TITLE
Relax concatenation checks for `--check_level=relax` and `--check_level=ignore`

### PR DESCRIPTION
**Pull request by sloosvel**
## Description

This PR allows to relax the concatenation checks for `--check_level=relaxed` and `--check_level=ignore`. Additionally, the concatenation is performed at once with all cubes, instead of in a two-by-two way. The overlaps in the time coordinate are checked before performing the concatenation, but they consider the same cases as the current implementation.

I only tested the concatenation of many files, but I don't know at the top of my head any data with the overlaps that are being considered. So I have not tested with any real dataset. The old tests seem to past though.

Closes (kind of) #2073 

Link to documentation: https://esmvaltool--2144.org.readthedocs.build/projects/ESMValCore/en/2144/

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
~- [ ] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/preprocessor/_io.py [hunk 1]
```diff
@@ -187,17 +188,110 @@ def load(file, callback=None, ignore_warnings=None):
...
+        'check_ancils': True,
+        'check_derived_coords': True
+    }
 
-    concatenated = _concatenate_overlapping_cubes(concatenated)
-    if len(concatenated) == 2:
-        _get_concatenation_error(concatenated)
-    else:
-        return concatenated[0]
+    if check_level > CheckLevels.DEFAULT:
```
Comment (valeriupredoi) [inline esmvalcore/preprocessor/_io.py:200]: isn't check_level a string? What's this semantic comparison do?

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
thanks very much @sloosvel - I am a bit confused by this PR but let me have a look at the entire code (in a branch checkout) first, then I'll post me comments :beer:

File: esmvalcore/preprocessor/_io.py (hunk 1 above)
Comment (sloosvel) [reply esmvalcore/preprocessor/_io.py:200]: It should be a CheckLevels instance, which sets a number to each level;

https://github.com/ESMValGroup/ESMValCore/blob/3222d28a8587e7ce3f5871522c035079b645dd5b/esmvalcore/cmor/check.py#L18

**Comment by bouweandela** [comment]
> One more point - `creation_date` may be a useful cube attribute for provenance/version tracking, do we have to remove it? 

This is recorded in our provenance here: https://github.com/ESMValGroup/ESMValCore/blob/ac31d5352e5dc79a3f75eba809069a8731c432ea/esmvalcore/_provenance.py#L205-L211
The attribute is removed from the cubes because it becomes invalid when we process the data.

File: esmvalcore/preprocessor/_io.py [hunk 2]
```diff
@@ -187,17 +188,110 @@ def load(file, callback=None, ignore_warnings=None):
...
-    if len(concatenated) == 1:
-        return concatenated[0]
+def _concatenate_cubes(cubes, check_level):
+    """Concatenate cubes according to the check_level."""
+    kwargs = {
+        'check_aux_coords': True,
+        'check_cell_measures': True,
+        'check_ancils': True,
+        'check_derived_coords': True
+    }
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_io.py:198]: Wouldn't it be nicer to just allow specifying these keyword arguments in the recipe as arguments to the preprocessor function `concatenate` and then pass them on to `CubeList.concatenate`? That is conceptually simpler and will give users more control.

Comment (bouweandela) [reply esmvalcore/preprocessor/_io.py:198]: Oh wait, that's no longer possible with the current design of `esmvalcore.dataset.Dataset.load`.

File: esmvalcore/preprocessor/_io.py (hunk 1 above)
Comment (valeriupredoi) [reply esmvalcore/preprocessor/_io.py:200]: gotcha, thanks @sloosvel

**Review by valeriupredoi — APPROVED** [review]
revising my stale review, cheers, Saskia! :beer:

**Pull request merged**